### PR TITLE
Optimize the way of generating XDR classes.

### LINKED ISSUE #502: Optimize the way of generating XDR classes.
### What problem does your feature solve?
dependent on manual execution of xdrgen, this offline step may be error-prone and does not lead to reproducible builds, since the commit hash/versions used at generation time for xdrgen and the xdr files themselves are are not captured.

See https://github.com/stellar/java-stellar-sdk/pull/498#discussion_r1286197370

### What would you like to see?

Add a Makefile build script similar to this on: https://github.com/stellar/go/blob/master/Makefile

### What alternatives are there?
N/A

### PR DESCRIPTION
**Pull request by overcat**
Closes #502

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Docker build targets use an optional "TAG" environment
```
Comment (sreuland) [inline Makefile:1]: not sure if the docs on TAG usage are relevant in here?

File: readme.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ For information on how to contribute, please refer to our [contribution guide](h
 java-stellar-sdk is licensed under an Apache-2.0 license. See the [LICENSE](https://github.com/stellar/java-stellar-sdk/blob/master/LICENSE) file for details.
 
 ## xdr to jave code generation
-All the java source files in org.stellar.sdk.xdr package are generated using the `xdrgen` command from the [stellar/xdrgen](https://github.com/stellar/xdrgen) 
-```
-xdrgen -o ./src/main/java/org/stellar/sdk/xdr -l java -n org.stellar.sdk.xdr ./xdr/*.x
+All the java source files in `org.stellar.sdk.xdr` package are generated using the following command:
```
Comment (sreuland) [inline readme.md:43]: nice!

**Review by sreuland — APPROVED** [review]
looks good, thanks for adding this tool/feature!

**Comment by overcat** [comment]
Ready for merge!

**Pull request merged**